lib/codegen/src/regalloc/virtregs.rs

```diff
@@ -291,20 +291,21 @@ impl UFEntry {
 impl VirtRegs {
     /// Find the leader value and rank of the set containing `v`.
     /// Compress the path if needed.
-    fn find(&mut self, val: Value) -> (Value, u32) {
-        match UFEntry::decode(self.union_find[val]) {
-            UFEntry::Rank(rank) => (val, rank),
-            UFEntry::Link(parent) => {
-                // TODO: This recursion would be more efficient as an iteration that pushes
-                // elements onto a SmallVector.
-                let found = self.find(parent);
-                // Compress the path if needed.
-                if found.0 != parent {
-                    self.union_find[val] = UFEntry::encode_link(found.0);
+    fn find(&mut self, mut val: Value) -> (Value, u32) {
+        let mut val_stack = vec![];
+        let found = loop {
+            match UFEntry::decode(self.union_find[val]) {
+                UFEntry::Rank(rank) => break (val, rank),
+                UFEntry::Link(new_val) => {
```

Comment:
Is there a reason for renaming `parent` to `new_val` here?

Reply:
I intend to show the relationship between `val` and `new_val`, that is, the old `val` is overwritten with the next value `new_val` in the loop.
Do you feel more natural to use `parent` instead of `new_val`?

Reply:
That makes sense, though I like `parent` here because the data structure forms a tree, and `parent` makes it clear that we're walking up the tree toward the root.